### ISSUE TITLE
False-positive of smallint type

### ISSUE THREAD

**Issue by SerafimArts** [phpstan/phpstan-doctrine#672]
The following code:

```php
    /**
     * @var int<-32768, 32767>
     */
    #[ORM\Column(name: 'sorting_order', type: 'smallint', options: ['default' => 0])]
    public int $order = 0;
```

Reports:
```
Property Entity::$order type mapping mismatch: database can contain int but property expects int<-32768, 32767>.
         🪪 doctrine.columnType
```

But expected:
```
OK
```

MySQL: https://dev.mysql.com/doc/refman/8.4/en/integer-types.html
> smallint is int<-32768, 32767>

PostgreSQL: https://www.postgresql.org/docs/current/datatype-numeric.html
> smallint is int<-32768, 32767>

Doctrine: https://www.doctrine-project.org/projects/doctrine-dbal/en/4.2/reference/types.html#smallint
> smallint is int<-32768, 32767>

**Comment by VincentLanglet** [comment]
It would require to update https://github.com/phpstan/phpstan-doctrine/blob/6fe9e9dfe6bb2fe6149c794ec3bb46658e9e9b64/src/Type/Doctrine/Descriptors/SmallIntType.php#L8 if you want to work on a PR @SerafimArts.

There is an `IntegerRangeType`.
